### PR TITLE
 NXP-28838 explorer json export (Part 2/2)

### PR DESCRIPTION
**Pull request by atchertchian**
Rest & UI follow-up of #12

### CODE REVIEW

File: modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/adapters/OperationInfoDocAdapter.java [hunk 1]
```diff
@@ -193,4 +196,14 @@ public String getContributingComponent() {
         return safeGet(PROP_CONTRIBUTING_COMPONENT);
     }
 
+    @Override
+    public ComponentInfo getComponent() {
+        DistributionSnapshot parentSnapshot = getParentNuxeoArtifact(DistributionSnapshot.class);
+        String cid = getContributingComponent();
+        if (parentSnapshot != null && !StringUtils.isEmpty(cid) && !BUILT_IN.equals(cid)) {
```
Comment (efge) [inline modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/adapters/OperationInfoDocAdapter.java:203]: You can use `StringUtils.isNotEmpty`

File: modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/introspection/RuntimeSnapshot.java [hunk 2]
```diff
@@ -392,17 +395,36 @@ protected void initOperations() {
...
                 documentation = op.getDocumentation();
             } catch (OperationException e) {
                 throw new NuxeoException(e);
             }
-            this.operations.add(new OperationInfoImpl(documentation, getVersion(), op.getType().getCanonicalName(),
-                    op.getContributingComponent()));
+            OperationInfo opi = new OperationInfoImpl(documentation, getVersion(), op.getType().getCanonicalName(),
+                    op.getContributingComponent());
+            String cid = opi.getContributingComponent();
+            if (!StringUtils.isBlank(cid) && !OperationInfo.BUILT_IN.equals(cid)) {
```
[the remark this answers is not in the document]
Comment (atchertchian) [reply modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/introspection/RuntimeSnapshot.java:409]: :+1: i went for "isNotBlank" still, given past inconsistencies on this information persistence

File: modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/snapshot/TargetExtensionPointSnapshotFilter.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        this.selectedArtifacts = selectedArtifacts;
+        index(selectedArtifacts);
+    }
+
+    protected void index(List<NuxeoArtifact> selectedArtifacts) {
+        // retrieve target extension points for given artifacts, assuming only complete bundles have been previously
+        // selected
+        List<ExtensionInfo> allExtensions = selectedArtifacts.stream()
+                                                             .filter(a -> a instanceof BundleInfo)
+                                                             .map(a -> (BundleInfo) a)
```
[the remark this answers is not in the document]
Comment (atchertchian) [reply modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/snapshot/TargetExtensionPointSnapshotFilter.java:62]: :+1: i actually used these later on and forgot to replace this one

**Comment by atchertchian** [comment]
Force-pushed to resolve conflicts after merge of #26

**Review by efge — APPROVED** [review]

**Comment by atchertchian** [comment]
Will need to force-push again to resolve conflict after merge of #29, so will merge as is right after.

Will probably open a last PR on this issue to prevent referencing requirements that would not be part of a partial distribution.

**Pull request merged**